WalletWasabi.Fluent/ViewModels/CoinJoinProfiles/CoinJoinProfilesViewModel.cs

```diff
@@ -55,15 +55,24 @@ private async Task OnManualSetupAsync()
 		}
 	}
 
-	private void OnNext(KeyManager keyManager)
+	private void OnNext(KeyManager keyManager, bool isNewWallet)
 	{
 		var selected = SelectedProfile ?? SelectedManualProfile ?? Profiles.First();
 
 		keyManager.AutoCoinJoin = selected.AutoCoinjoin;
 		keyManager.SetAnonScoreTargets(selected.MinAnonScoreTarget, selected.MaxAnonScoreTarget, toFile: false);
 		keyManager.SetFeeRateMedianTimeFrame(selected.FeeRateMedianTimeFrameHours, toFile: false);
+		keyManager.IsCoinjoinProfileSelected = true;
 
-		Navigate().To(new AddedWalletPageViewModel(keyManager));
+		keyManager.ToFile();
```

Comment:
Should this be inside of the `else` branch? `AddedWalletPageViewModel` saves the wallet as well,

Reply:
If we follow the call stack after `AddedWalletPageViewModel` , we see `ToFile()` being called in `KeyManager.cs`, at L568 in `AssertCleanKeysIndexed()`. According to @molnard we shouldn't rely on saving to file during generating `HdPubKeys`. Instead we should make sure to save the changes after them have been made.

Reply:
I see. But the last page should be responsible for the saving.

It is not an issue now but imagine that in the future there would be another page between the coinjoin profile set and the added wallet page. The user could press cancel on the new page so the adding should be canceled but the wallet will appear in the software because the saving was called not on the last page.